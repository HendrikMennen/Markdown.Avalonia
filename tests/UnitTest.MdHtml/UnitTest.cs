using ApprovalTests;
using ApprovalTests.Reporters;
using Avalonia.Controls;
using Markdown.Avalonia;
using Markdown.Avalonia.Html;
using NUnit.Framework;
using System.Linq;
using UnitTest.Base;
using UnitTest.Base.Utils;
using UnitTest.MdHtml.Test;

using MarkdownEngine = Markdown.Avalonia.Markdown;

namespace UnitTest.MdHtml
{
    [UseReporter(typeof(DiffReporter))]
    public class UnitTest : UnitTestBase
    {
        public MarkdownEngine Engine
        {
            get
            {
                var plugins = new MdAvPlugins();
                plugins.Plugins.Add(new HtmlPlugin());

                return new MarkdownEngine()
                {
                    Plugins = plugins
                };
            }
        }

        [Test]
        [RunOnUI]
        public void Button()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void CodeBlock()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void InlineCode()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void Input()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void List()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void Progres()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void Details()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void TypicalBlock()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void TypicalInline()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void Mixing()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }
    }
}
