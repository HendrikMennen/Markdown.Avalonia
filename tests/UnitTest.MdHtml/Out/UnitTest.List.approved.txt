<?xml version="1.0" encoding="utf-16"?>
<StackPanel xmlns:c="clr-namespace:ColorTextBlock.Avalonia;assembly=ColorTextBlock.Avalonia" xmlns:m="https://github.com/whistyun/Markdown.Avalonia.Tight" xmlns:ma="clr-namespace:Markdown.Avalonia.Controls;assembly=Markdown.Avalonia" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns="https://github.com/avaloniaui">
  <Grid xmlns="">
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="0">
      <c:CRun Text="1." />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="0">
      <c:CTextBlock>
        <c:CRun Text="asdf " />
        <c:CBold Classes="Bold">
          <c:CRun Text="fdsa " />
        </c:CBold>
        <c:CRun Text=" jkl; " />
        <c:CItalic Classes="Italic">
          <c:CRun Text=";lkj" />
        </c:CItalic>
      </c:CTextBlock>
    </StackPanel>
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="1">
      <c:CRun Text="2." />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="1">
      <c:CTextBlock>
        <c:CRun Text="1234567" />
      </c:CTextBlock>
    </StackPanel>
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="2">
      <c:CRun Text="3." />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="2">
      <c:CTextBlock>
        <c:CInlineUIContainer>
          <c:CInlineUIContainer.Content>
            <RadioButton IsEnabled="False" Classes=":unchecked,:disabled" />
          </c:CInlineUIContainer.Content>
        </c:CInlineUIContainer>
        <c:CRun Text="ねむい" />
      </c:CTextBlock>
    </StackPanel>
    <Grid.ColumnDefinitions>
      <ColumnDefinition />
      <ColumnDefinition />
    </Grid.ColumnDefinitions>
    <Grid.RowDefinitions>
      <RowDefinition />
      <RowDefinition />
      <RowDefinition />
    </Grid.RowDefinitions>
  </Grid>
  <ma:Rule BoldLineWidth="2" Classes="Single,RuleSingle" />
  <Grid xmlns="">
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="0">
      <c:CRun Text="・" />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="0">
      <c:CTextBlock>
        <c:CRun Text="asdf " />
        <c:CBold Classes="Bold">
          <c:CRun Text="fdsa " />
        </c:CBold>
        <c:CRun Text=" jkl; " />
        <c:CItalic Classes="Italic">
          <c:CRun Text=";lkj" />
        </c:CItalic>
      </c:CTextBlock>
    </StackPanel>
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="1">
      <c:CRun Text="・" />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="1">
      <c:CTextBlock>
        <c:CRun Text="1234567" />
      </c:CTextBlock>
    </StackPanel>
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="2">
      <c:CRun Text="・" />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="2">
      <c:CTextBlock>
        <c:CInlineUIContainer>
          <c:CInlineUIContainer.Content>
            <RadioButton IsEnabled="False" Classes=":unchecked,:disabled" />
          </c:CInlineUIContainer.Content>
        </c:CInlineUIContainer>
        <c:CRun Text="ねむい" />
      </c:CTextBlock>
    </StackPanel>
    <Grid.ColumnDefinitions>
      <ColumnDefinition />
      <ColumnDefinition />
    </Grid.ColumnDefinitions>
    <Grid.RowDefinitions>
      <RowDefinition />
      <RowDefinition />
      <RowDefinition />
    </Grid.RowDefinitions>
  </Grid>
  <ma:Rule BoldLineWidth="2" Classes="Single,RuleSingle" />
  <Grid xmlns="">
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="0">
      <c:CRun Text="12." />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="0">
      <c:CTextBlock>
        <c:CRun Text="asdf " />
        <c:CBold Classes="Bold">
          <c:CRun Text="fdsa " />
        </c:CBold>
        <c:CRun Text=" jkl; " />
        <c:CItalic Classes="Italic">
          <c:CRun Text=";lkj" />
        </c:CItalic>
      </c:CTextBlock>
    </StackPanel>
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="1">
      <c:CRun Text="13." />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="1">
      <c:CTextBlock>
        <c:CRun Text="1234567" />
      </c:CTextBlock>
    </StackPanel>
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="2">
      <c:CRun Text="14." />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="2">
      <c:CTextBlock>
        <c:CInlineUIContainer>
          <c:CInlineUIContainer.Content>
            <RadioButton IsEnabled="False" Classes=":unchecked,:disabled" />
          </c:CInlineUIContainer.Content>
        </c:CInlineUIContainer>
        <c:CRun Text="ねむい" />
      </c:CTextBlock>
    </StackPanel>
    <Grid.ColumnDefinitions>
      <ColumnDefinition />
      <ColumnDefinition />
    </Grid.ColumnDefinitions>
    <Grid.RowDefinitions>
      <RowDefinition />
      <RowDefinition />
      <RowDefinition />
    </Grid.RowDefinitions>
  </Grid>
</StackPanel>
